tests/Unit/TestHelpers.hpp

```diff
@@ -393,3 +397,128 @@ struct check_variables_approx<Variables<TagList>> {
     });
   }
 };
+
+/*!
+ * \ingroup TestingFrameworkGroup
+ * \brief A test utility for verifying that an element-wise function, `function`
+ * acts identically to the same operation applied to each element of a container
+ * separately. This macro invokes `test_element_wise_function()` (which gives a
+ * more complete documentation of the element-wise checking operations and
+ * arguments).
+ */
+#define CHECK_ELEMENT_WISE_FUNCTION_APPROX(function, arguments) \
+  do {                                                          \
+    INFO(__FILE__ ":" + std::to_string(__LINE__) +              \
+         ": " #function ", " #arguments);                       \
+    test_element_wise_function(function, arguments);            \
+  } while (false)
+
+/*!
+ * \ingroup TestingFrameworkGroup
+ * \brief Same as `CHECK_ELEMENT_WISE_FUNCTION_APPROX`, but with a user-defined
+ * function `at_operator` and `size_of_operator`, each of which correspond to
+ * arguments of `test_element_wise_function()` (which gives a more complete
+ * documentation of the element-wise checking operations and arguments).
+ */
+#define CHECK_CUSTOM_ELEMENT_WISE_FUNCTION_APPROX(               \
+    function, arguments, at_operator, size_of_operator)          \
+  do {                                                           \
+    INFO(__FILE__ ":" + std::to_string(__LINE__) +               \
+         ": " #function ", " #arguments);                        \
+    test_element_wise_function(function, arguments, at_operator, \
+                               size_of_operator);                \
+  } while (false)
+
+namespace TestHelpersDetail {
+// CHECK forwarding for parameter pack expansion. Return value also required for
+// easy parameter pack use.
+SPECTRE_ALWAYS_INLINE int call_check(const bool input) noexcept {
+  CHECK(input);
+  return 0;
+}
+
+// internal implementation for checking that an element-wise function acts
+// appropriately on a container object by first trying the operation on the
+// containers, then checking that the operation has been performed as it would
+// by looping over the elements and performing the same operation on each. This
+// also supports testing functions between containers and single elements by
+// making use of `get_size` and `get_element`.
+// the arguments tuple is taken by mutable reference to support inplace operator
+// testing.
+template <typename Function, typename IndexingFunction, typename SizeFunction,
+          typename... Arguments, size_t... Is>
+void test_element_wise_function_impl(Function elementwise_function,
+                                     std::tuple<Arguments...>& arguments,
```

Comment:
since the `arguments` are being modified they should be taken by `gsl::not_null`